src/gluonts/model/deeprenewal/_network.py

```diff
@@ -0,0 +1,629 @@
+            I = len(indices), containing lagged subsequences. Specifically,
+            lagged[i, j, :, k] = sequence[i, -indices[k]-S+j, :].
+        """
+        # we must have: sequence_length - lag_index - subsequences_length >= 0
+        # for all lag_index, hence the following assert
+        assert max(indices) + subsequences_length <= sequence_length, (
+            f"lags cannot go further than history length, "
+            f"found lag {max(indices)} while history length is only "
+            f"{sequence_length}"
+        )
+        assert all(lag_index >= 0 for lag_index in indices)
+
+        lagged_values = []
+        for lag_index in indices:
+            begin_index = -lag_index - subsequences_length
+            end_index = -lag_index if lag_index > 0 else None
+            lagged_values.append(
+                F.slice_axis(sequence, axis=1, begin=begin_index, end=end_index)
+            )
+
+        return F.stack(*lagged_values, axis=-1)
+
+    def unroll_encoder(
+        self,
+        F,
+        feat_static_cat: Tensor,  # (batch_size, num_features)
+        feat_static_real: Tensor,  # (batch_size, num_features)
+        past_time_feat: Tensor,  # (batch_size, history_length, num_features)
+        past_target: Tensor,  # (batch_size, history_length, *target_shape)
+        past_observed_values: Tensor,  # (batch_size, history_length, *target_shape)
+        future_time_feat: Optional[
+            Tensor
+        ],  # (batch_size, prediction_length, num_features)
+        future_target: Optional[
+            Tensor
+        ],  # (batch_size, prediction_length, *target_shape)
+    ) -> Tuple[Tensor, List, Tensor, Tensor]:
+        """
+        Unrolls the LSTM encoder over past and, if present, future data.
+        Returns outputs and state of the encoder, plus the scale of past_target
+        and a vector of static features that was constructed and fed as input
+        to the encoder.
+        All tensor arguments should have NTC layout.
+        """
+
+        if future_time_feat is None or future_target is None:
+            time_feat = past_time_feat.slice_axis(
+                axis=1, begin=self.history_length - self.context_length, end=None,
+            )
+            sequence = past_target
+            sequence_length = self.history_length
+            subsequences_length = self.context_length
+        else:
+            time_feat = F.concat(
+                past_time_feat.slice_axis(
+                    axis=1, begin=self.history_length - self.context_length, end=None,
+                ),
+                future_time_feat,
+                dim=1,
+            )
+            sequence = F.concat(past_target, future_target, dim=1)
+            sequence_length = self.history_length + self.prediction_length
+            subsequences_length = self.context_length + self.prediction_length
+
+        # (batch_size, sub_seq_len, *target_shape, num_lags)
+        lags_m = self.get_lagged_subsequences(
```

Comment:
Does the network need lags? I believe in the original paper we don't have lags as input, but rely only on the LSTM output.

Reply:
I agree.. Lags are an overkill here.. we can remove it..